README.md

```diff
@@ -1,51 +1,98 @@
+username = "your_kik_username"
+password = "your_kik_password"
+
+# The bot class that handles all the callbacks from the kik client
 class EchoBot(KikClientCallback):
     def __init__(self):
-        self.client = KikClient(self, "your_kik_username", "your_kik_password")
+        self.client = KikClient(self, username, password)
         self.client.wait_for_messages()
 
+    # This method is called when the bot is authenticated
     def on_authenticated(self):
         self.client.request_roster() # request list of chat partners
 
+    # This method is called when the bot receives a chat message in your dms
     def on_chat_message_received(self, chat_message: chatting.IncomingChatMessage):
-        self.client.send_chat_message(chat_message.from_jid, 'You said "{}"!'.format(chat_message.body))
+        self.client.send_chat_message(chat_message.from_jid, f'You said "{chat_message.body}"!')
+    
+    # This method is called when the bot receives a chat message in a group
+    def on_group_message_received(self, chat_message: chatting.IncomingGroupChatMessage):
+        self.client.send_chat_message(chat_message.from_jid, f'You said "{chat_message.body}"!')
+    
+    # This method is called if a captcha is required to login
+    def on_login_error(self, login_error: LoginError):
+        if login_error.is_captcha():
+            login_error.solve_captcha_wizard(self.client)
+
+
+if __name__ == '__main__':
+    # Creates the bot and start listening for incoming chat messages
+    callback = EchoBot()
+    client = KikClient(callback=callback, kik_username=username, kik_password=password)
+    client.wait_for_messages()
+        
 ```
+Please replace "your_kik_username" and "your_kik_password" with your actual Kik username and password.
 
-Currently Supported Operations:
-- Log in with kik username and password, retrieve user information (such as email, name, etc).
-- Fetch chat partners information
-- Send text messages to users/groups and listen for incoming messages
-- Send and receive 'is-typing' status
-- Send and receive read receipts
-- Fetch group information (name, participants, etc.)
-- Fetch past message history
-- Admin groups (add, remove or ban members, etc)
-- Search for groups and join them [Experimental]
-- Receive media content: camera, gallery, stickers
-- Add a kik user as a friend
-- Send images (including GIFs, using a [Tenor](https://tenor.com/gifapi) API key)
+You can run this example by running `python3 examples/simple_echo_bot.py`. Visit the [examples](examples) directory for more examples.
+
+## Captcha Solving ##
+Once the bot starts running, you might see a message like this:
+`To continue, complete the captcha in this URL using a browser: https://captcha.kik.com/?id=...`
+
+
+This means that Kik has detected that you are using a bot and requires you to solve a captcha to continue. You can solve the captcha by opening the URL in a browser and following these steps:
+
+- Press f12 to open the developer tools
```

Comment:
`F12` instead of `f12`

Reply:
Also these instructions are printed when the captcha wizard starts, do we really need them here?

Reply:
The captcha wizard isn't super clear. Here is what it says:

To continue, complete the captcha in this URL using a browser: https://captcha.kik.com/?id=...
Next, intercept the request starting with 'https://kik.com/captcha-url' using F12, and paste the response parameter here: 

I don't think most people know what "intercept the request" means unless they've done it before

Reply:
OK, I agree